### PR TITLE
Add spring cloud template

### PR DESCRIPTION
**Pull request by mgblythe**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

File: 101-azure-spring-cloud/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        },
+        "app3": {
+            "type": "string",
+            "defaultValue": "auth-service"
+        },
+        "location": {
+          "type": "string",
+          "defaultValue": "[resourceGroup().location]"
+        },
+        "location-spring-cloud": {
```
Comment (bmoore-msft) [inline 101-azure-spring-cloud/azuredeploy.json:24]: remove this and just use the location parameter... do not add the defaultValue, set it in the parameter file.

File: 101-azure-spring-cloud/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          "type": "string",
+          "defaultValue": "westus2",
+          "allowedValues": ["westeurope","eastus","westus2","southeastasia","centralus","australiaeast","uksouth","northeurope","southcentralus","eastus2"]
```
Comment (bmoore-msft) [inline 101-azure-spring-cloud/azuredeploy.json:27]: format the json, e.g. SHIFT+ALT+F in VS Code

File: 101-azure-spring-cloud/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+![Azure US Gov Last Test Result](https://azurequickstartsservice.blob.core.windows.net/badges/101-azure-spring-cloud/FairfaxDeployment.svg)
+
+![Best Practice Check](https://azurequickstartsservice.blob.core.windows.net/badges/101-azure-spring-cloud/BestPracticeResult.svg)
+![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/101-azure-spring-cloud/CredScanResult.svg)
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-azure-spring-cloud%2Fazuredeploy.json)
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-azure-spring-cloud%2Fazuredeploy.json)
```
Comment (bmoore-msft) [inline 101-azure-spring-cloud/README.md:13]: remove this since the sample is not supported in FF

File: 101-azure-spring-cloud/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                                    "type": "Jar"
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
```
Comment (bmoore-msft) [inline 101-azure-spring-cloud/azuredeploy.json:134]: add some explanation to the readme.md to explain why we're deploying these twice.

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-azure-spring-cloud/azuredeploy.json (hunk 4 above)
Comment (mgblythe) [reply 101-azure-spring-cloud/azuredeploy.json]: Will wait for review/input from the product team, unless you know why it's deployed twice

**Comment by mgblythe** [comment]
@bmoore-msft - I made requested changes except as noted with the second deployment. The product team reviewer is OOF this week, so I'll follow up next week and finalize.

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by bmoore-msft** [comment]
@mgblythe - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Comment by mgblythe** [comment]
@bmoore-msft - do you know why USGocv would be failing now given that I've excluded it in the metadata file?

File: 101-azure-spring-cloud/azuredeploy.json (hunk 4 above)
Comment (yucwan) [reply 101-azure-spring-cloud/azuredeploy.json]: "name": "setActiveDeployment",
As the name indicates, the second time to deploy is to set the "deployments" as "active" for the corresponding apps. Before doing this all resource are provisioned but traffic to apps is not routed to the running "deployments".

**Pull request merged**